### DISCUSSION TITLE
Setting up a proxy forward to my Synology NAS hosted instance

### DISCUSSION THREAD

**Discussion by TimToxopeus** [felddy/foundryvtt-docker#228] [Q&A] [answered]
Hi, I have been running Foundry just fine on my Synology NAS for the past year or so, but I always accessed it with my public ip and port. I want to enable the webcam support, and as far as I understand for that I need to set up a certificate, and to set up a certificate I need to have a domain to set it up for.

So I went to my webhost, added a subdomain, and made it a proxy forward to my Foundry IP/Port. When I access it in browser, I get the landing page, but nothing loads otherwise. In the console I get a bunch of the following error:
`WebSocket connection to 'ws://my.domain.com/socket.io/?session=0d46ec70939756051e234611&EIO=4&transport=websocket' failed: 
`

I probably need to configure something somewhere, but I am not sure how to do it. In the docker-compose.yml file I set hostname to my.domain.com, and added environment variables for FOUNDRY_HOSTNAME and FOUNDRY_LOCAL_HOSTNAME and set them both to my.domain.com as well. But this didn't resolve the problem.

I tried playing with setting FOUNDRY_PROXY_PORT to 80 and 30000, to see if that made a difference, but it only caused for the server to be inaccessible (HTTP 502 Proxy Error).

I saw some random things online about it, this thread being the closest:
https://www.foundryvtt-hub.com/community/foundry-core-how-to/foundryvtt-behind-reverse-proxy-websockets-not-working/?orderby=newest
But that is obviously talking about an entirely different hosting solution and I couldn't immediately figure out how to apply any of the concepts there to my docker image.

**Comment by TimToxopeus** [comment, marked as answer]
So, thanks to the help of STB from the Foundry discord, this has been resolved. I had to tackle a bunch of different steps:
1. On my webhost, I had to disable forward proxy, and instead make an A-Record entry pointing directly to my NAS' public IP address.
2. On my router, I had to port forward 80 and 443 to my NAS.
3. On my NAS, I had to set up a Reverse Proxy Server. (See https://kb.synology.com/en-global/DSM/help/DSM/AdminCenter/application_appportalias?version=6 )
3b. On the Reverse Proxy entry, you can also add Custom Headers, if you go to that tab, the Create button has a dropdown box for Websocket which you can click to automatically fill in websocket entries.
4. On my NAS, I also had to set up a self-signed certificate, and associate it with the Reverse Proxy entries.
5. In my docker-compose.yml file I set the hostname to the appropriate domain.
6. Then after restarting docker images, it works. :)

**Reply by felddy** [reply]
Great success!  I'm glad you got this working.  I apologize for not being able to help troubleshoot.  

Also, thanks for posting your solution to help the people from the future! 


![XKCD 979](https://imgs.xkcd.com/comics/wisdom_of_the_ancients.png)
https://m.xkcd.com/979/

**Reply by davidmichelruddy** [reply]
This solved it for me too on my Synology DS920+. Specifically 3b above.

> 3b. On the Reverse Proxy entry, you can also add Custom Headers, if you go to that tab, the Create button has a dropdown box for Websocket which you can click to automatically fill in websocket entries.

I had everything else dialed in and this was the missing piece. Thank you!
